src/node/constants.ts

```diff
@@ -18,3 +19,4 @@ const pkg = getPackageJson("../../package.json")
 export const version = pkg.version || "development"
 export const commit = pkg.commit || "development"
 export const rootPath = path.resolve(__dirname, "../..")
+export const tmpdir = path.join(os.tmpdir(), "code-server")
```

Comment:
💄 Moved here for aesthetics. It is a constant and never changes

Reply:
Always love your use of emoji, this put a smile on my face Joe!

Based on a quick read of the docs, `os.tmpdir()` returns an a-priori known directory (e.g. /tmp) so there are potential security issues, since this effectively convinces the test to write to a directory that we don't directly control. Probably not a problem since these are tests though. Would it make sense to add an unpredictable part to this, e.g. `path.join(os.tmpdir(), "code-server" + current pid)`?

Reply:
What about something like this for tests (and we move it into `test/utils/constants.ts`):

```
export async function tmpdir(testName: string): Promise<string> {
  const dir = path.join(os.tmpdir(), "code-server")
  await fs.mkdir(dir, { recursive: true })
  return fs.mkdtemp(path.join(dir, `test-${name}-`))
}
```

`tmpdir` is used in one place outside of tests when creating the socket proxy but I think maybe we should use the data directory instead of using tmp and we should randomize it. Would fix https://github.com/cdr/code-server/issues/2200#issuecomment-709460909

Reply:
TIL about the potential security issues!

Totally I will add that to `constants`.

Re: changing to `data` directory feels out of scope of this PR, but will make a note and do in a followup PR